pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/JdbcUtils.java

```diff
@@ -94,7 +94,7 @@ public static TableDefinition of(TableId tableId, List<ColumnId> columns,
      */
     public static TableId getTableId(Connection connection, String tableName) throws Exception {
         DatabaseMetaData metadata = connection.getMetaData();
-        try (ResultSet rs = metadata.getTables(null, null, tableName, new String[]{"TABLE"})) {
+        try (ResultSet rs = metadata.getTables(connection.getCatalog(), null, tableName, new String[]{"TABLE"})) {
```

Comment:
```suggestion
        try (ResultSet rs = metadata.getTables(connection.getCatalog(), connection.getSchema(), tableName, new String[]{"TABLE"})) {
```

? I think this can be trivial to merge as long as all tests passed.

Reply:
It's not quite easy to add test case without involving external db systems.